lib/devicetrust/native/api.go

```diff
@@ -34,24 +41,77 @@ const (
 	// CollectedDataNeverEscalate will never escalate privileges, even in the
 	// absence of cached data.
 	CollectedDataNeverEscalate CollectDataMode = iota
-	// CollectedDataAlwaysEscalate avoids using cached DMI data and instead will
-	// always escalate privileges if necessary.
-	//
-	// Used by `tsh device enroll`, `tsh device collect` and
-	// `tsh device asset-tag`.
-	CollectedDataAlwaysEscalate
 	// CollectedDataMaybeEscalate will attempt to use cached DMI data before
 	// privilege escalation, but it may choose to escalate if no cached data is
 	// available.
 	//
 	// Used by `tsh login` and similar operations (ie, device authn).
 	CollectedDataMaybeEscalate
+	// CollectedDataAlwaysEscalate avoids using cached DMI data and instead will
+	// always escalate privileges if necessary.
+	//
+	// Used by `tsh device enroll`, `tsh device collect` and
+	// `tsh device asset-tag`.
+	CollectedDataAlwaysEscalate
+	// IMPORTANT: CollectDataMode declarations must go from least to most strict.
 )
 
+var cachedDeviceData = struct {
```

Comment:
Did you consider using our FnCache utility? Might reduce some boilerplate.

Reply:
I think I'd rather keep this, it lets me deal cleanly with the `cache.mode < mode` logic.